app/core/Tracker2/DriftTracker.js

```diff
@@ -72,11 +83,40 @@ export default class DriftTracker extends BaseTracker {
     // Show when a message is received
     window.drift.on('message', (e) => {
       if (!e.data.sidebarOpen) {
-        this.driftApi.widget.show()
+        if (this.isChatEnabled) {
+          this.driftApi.widget.show()
+        }
       }
     })
   }
 
+  routeUpdated () {
+    this.updateDriftConfiguration()
+  }
+
+  userRoleUpdated () {
+    this.updateDriftConfiguration()
+  }
+
+  get onPlayPage () {
+    const { route } = this.store.state
+    return (route.path || '').indexOf('/play/') === 0
+  }
+
+  get isChatEnabled () {
+    return !this.onPlayPage && !this.store.getters['me/isStudent']
```

Comment:
Have to call the function, right?
```suggestion
    return !this.onPlayPage() && !this.store.getters['me/isStudent']
```

Reply:
Or maybe I'm misreading it and the getter takes care of this?

Reply:
Getter should take care of this, but a great call out. I also had to remind myself. Great use of modern features Joe!

```
class Rawr {
  get hey() {
    console.log('side effect')
  }
}

>> let r = new Rawr()

>> r.hey
undefined
// -> side effect
```